src/core/ddsi/src/ddsi_security_omg.c

```diff
@@ -1714,6 +1714,15 @@ unsigned determine_publication_writer (const struct writer *wr)
     return NN_ENTITYID_SEDP_BUILTIN_PUBLICATIONS_WRITER;
 }
 
+#ifdef DDS_HAS_TOPIC_DISCOVERY
+unsigned determine_topic_writer (const struct topic *tp)
+{
+  // FIXME: secure topic writer required?
```

Comment:
I've changed this so that the implementation will `abort` in case topic discovery is used (not only enabled the build option, but also the topic discovery endpoints enabled in config) and discovery protection is enabled in the participant.